tests/pyincore/utils/test_pdcsvoutputjson.py

```diff
@@ -0,0 +1,65 @@
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+from pyincore.utils.huapdoutputprocess import HUADislOutputProcess
+from pyincore import IncoreClient, FragilityService, MappingSet
+from pyincore.analyses.populationdislocation import PopulationDislocation
+import pyincore.globals as pyglobals
+import os
+
+def run_convert_pd_json_chained():
+    client = IncoreClient(pyglobals.INCORE_API_DEV_URL)
+
+    # Joplin population dislocation
+    # incore-dev
+    building_dmg = "602d96e4b1db9c28aeeebdce"  # dev Joplin
+    # building_dmg = "602d975db1db9c28aeeebe35" # 15 guids test - dev Joplin
+    housing_unit_alloc = "61563545483ecb19e4304c2a"  # dev Joplin 2ev3
+    bg_data = "5df7cb0b425e0b00092c9464"  # Joplin 2ev2
+    value_loss = "602d508fb1db9c28aeedb2a5"
+
+    result_name = "joplin-pop-disl-results"
+    seed = 1111
+
+    pop_dis = PopulationDislocation(client)
+
+    pop_dis.load_remote_input_dataset("building_dmg", building_dmg)
+    pop_dis.load_remote_input_dataset("housing_unit_allocation", housing_unit_alloc)
+    pop_dis.load_remote_input_dataset("block_group_data", bg_data)
+    pop_dis.load_remote_input_dataset("value_poss_param", value_loss)
+
+    pop_dis.set_parameter("result_name", result_name)
+    pop_dis.set_parameter("seed", seed)
+
+    pop_dis.run_analysis()
+    pd_result = pop_dis.get_output_dataset("result")
+
+    pd_json = HUADislOutputProcess(pd_result)
+    return pd_json
+
+def run_convert_pd_json_path(testpath):
+    # test the external file with a path
+    pd_json = HUADislOutputProcess(None, os.path.join(testpath, "joplin-pop-disl-results.csv"))
+    return pd_json
+
+
+if __name__ == '__main__':
+    # test chaining with population dislocation
+    pd_json = run_convert_pd_json_chained()
+
+    pd_json.pd_by_race("PD_by_race.json")
+    pd_json.pd_by_income("PD_by_income.json")
+    pd_json.pd_by_tenure("PD_by_tenure.json")
+    pd_json.pd_by_housing("PD_by_housing.json")
+    pd_json.pd_total("PD_by_total.json")
+
```

Comment:
@navarroc you can use this snippet on the workflow

Reply:
I would need an RC build of pyincore so I can add the util method. We can do that once this is merged